File: Mirea/Алгоритмы и структуры данных/Практика 1.2/1.1.py
def binary_search(arr, x):
    l, r = 0, len(arr) - 1
    while l <= r:
        if x == arr[(r + l) // 2]: return (r + l) // 2
        elif x < arr[(r + l) // 2]: r = (l + r) // 2
        else: l = (r + l) // 2
        if r - l == 1: return r, arr[:r] + [x] + arr[r:]


a = [1, 2, 3, 5, 6, 7, 9]
print(binary_search(a, 7))
print(binary_search(a, 4))
print(binary_search(a, 8))

File: Mirea/Алгоритмы и структуры данных/Практика 1.2/1_1.cpp
#include <iostream>
#include <array>
using namespace std;

int binar(int *arr,int x){
    int l(0);
    int r(sizeof(arr) - 2);
    while (l <= r){
        int q((r + l) / 2);
        cout << q << " " << l << " " << r << "\n";
        if (x == arr[q]){
            return q;
        }
        else if (x < arr[q]){
            r = q;
        }
        else{
            l = q;
        }
        if (r - l == 1){
            return l;
        }
    }
    return 0;
}

int main(){
    int values[]{1,2,3,5,6,7};
    cout <<  - 1 << "\n";
    /*cout << binar(values, 7);
    cout << binar(values, 4);
    cout << binar(values, 8);*/
}